projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -98,38 +100,81 @@ async function createOrUpdateComment( payload, octokit, issueReferences, issueCo
 	const { name: repo, owner } = repository;
 	const ownerLogin = owner.login;
 
-	const comment = `**Support References**
-	${ issueReferences
-		.map(
-			reference => `
-- [ ] ${ reference }`
-		)
-		.join( '' ) }
-
-	`;
-
-	const commentOpts = {
-		owner: ownerLogin,
-		repo,
-		body: comment,
-	};
-
 	const existingComment = await getListComment( issueComments );
 
 	// If there is a comment already, update it.
 	if ( existingComment !== 0 ) {
 		debug(
 			`gather-support-references: update comment ID ${ existingComment } with our new list of references.`
 		);
+
+		const {
+			data: { body: existingCommentBody },
+		} = await octokit.rest.issues.getComment( {
+			owner: ownerLogin,
+			repo,
+			comment_id: existingComment,
+		} );
+
+		// First, build a list of all references and whether they are checked or not.
+		const listWithStatusMatch = existingCommentBody.matchAll(
+			/(?:-\s\[(?<checked>\s|x)\]\s)(?<ticketId>[0-9]*-(?:hc|zen))/gm
```

Comment:
I wonder if we should loosen the ticketid part up to reduce future maintenance burden. Or at least cross-reference this and `referencesRegexP` above with some comments so future-us remember to update both if we add new recognized references.

On the other hand, unless we're worried about people manually editing the bot's comment and screwing up the formatting we could probably get away with `\S+` in here to grab anything on the line.

Also, BTW, you have an unnecessary non-capturing group, and we might want to anchor it.
```suggestion
			/^-\s\[(?<checked>\s|x)\]\s(?<ticketId>\S+)/gm
```

Reply:
Actually, come to think of it, you don't really need the "checked" group. Just match for `/^-\s\[x\]\s(?<ticketId>\S+)/gm` and process all the matches.

Reply:
Good point. e6a12e367df4f4389d5352de1f5dade6f73f20d9 should do it I think.